### ISSUE TITLE
400 Error When Attempting to Log In

### ISSUE THREAD

**Issue by joerhoney** [commitspark/frontend#4]
I LOVE this concept and I'm glad someone is tackling it! I think this project has a great deal of potential.

I followed the steps in your instructions and I now get a Commitspark login screen when I run `npm run dev`. I can log in and see my repos. Here's my feedback and issue(s) (maybe I should break these down into separate issues, or maybe some of them are easily addressed here):

1. I can log in, but I'm seeing ALL of my repos, even though I set the GitHub app to only allow access to the one I intend to manage content with. Is that expected, since they're my own repos? Can I set it to only show a specific one?
2. When I click on a repo, I can see the branches, but no content types are displayed. It appears to be attempting to load. Based on your docs, I assume that's because I don't have a GraphQL schema file in place yet. I think I'm suggesting having some sort of notification or error message display if no content types can be found, and/or perhaps a default GraphQL schema in your starter repo (but maybe there's a reason you don't have that).

Hope you don't mind me posting this issue. Again, great work!

**Comment by advancingu** [comment]
Hi @joerhoney, Thanks for the praise!

Here are my thoughts on your points:

1. The underlying GitHub API call is [this one](https://github.com/commitspark/frontend/blob/main/lib/provider/github/github-provider.ts#L24). As far as I understand it and based on [GitHub's API documentation](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repositories-for-the-authenticated-user), this call returns all of the following repositories:
    * All repositories located directly on your user account independent of visibility
    * All public organization repositories that you have access to
    * Private organization repositories that you gave access to via the GitHub App configuration
    I haven't looked deeper into the query to see if/how it is possible to limit results to only those explicitly enabled in the App. For now, the behavior you are expecting is only possible with private organization-owned repositories. I guess I should also check if the results could be filtered to those that the app is authorized for.
2. The frontend is still missing a lot of functionality in terms of user-friendly error handling and that needs to be expanded significantly. Currently you should however at least see errors show up on the JavaScript console in your browser.
    The behavior you're seeing is correct though if there is no schema file at the expected location in the repository (and branch) you're looking at. I recommend you to simply copy the template repository https://github.com/commitspark/example-content-website and use it as a sandbox for experimenting.

Would be happy if you could spread the word about this project. Lack of visibility is currently my biggest non-technical issue. Thanks!

**Comment by joerhoney** [comment]
Definitely! I'd love for more people to see the incredible value in Commitspark. 

Okay, makes sense and very helpful insight. Thank you! Sounds like there is just more work to be done on those points, but I will watch my console. And maybe if the app is made on an account specific to an organization (as opposed to a personal account as I'm trying it out on), seeing all of the repositories isn't a big deal in most cases.

Thanks for the reply! I've updated the name of this issue to be more relevant and I'm closing it out (at least for now) on the basis that it is resolved for me. Let me know if you'd like me to reopen or change its status for any reason.
